siriuspy/siriuspy/idff/main.py

```diff
@@ -0,0 +1,226 @@
+        return self._pvs_database
+
+    def process(self, interval):
+        """Sleep."""
+        _time.sleep(interval)
+
+    def read(self, reason):
+        """Read from IOC database."""
+        value = None
+        return value
+
+    def write(self, reason, value):
+        """Write value to reason and let callback update PV database."""
+        _log.info('Write received for: %s --> %s', reason, str(value))
+        if reason in self.map_pv2write.keys():
+            status = self.map_pv2write[reason](value)
+            _log.info('%s Write for: %s --> %s',
+                      str(status).upper(), reason, str(value))
+            return status
+        _log.warning('PV %s does not have a set function.', reason)
+        return False
+
+    def set_loop_state(self, value):
+        """Set loop state."""
+        if not 0 <= value < len(_ETypes.OPEN_CLOSED):
+            return False
+        self._loop_state = value
+        act = ('En' if value else 'Dis')
+        self._update_log(f'Feedforward loop {act}abled.')
+        self.run_callbacks('LoopState-Sts', value)
+        return True
+
+    def set_loop_freq(self, value):
+        """Set loop frequency."""
+        if not 1e-3 <= value < 60:
+            return False
+        self._loop_freq = value
+        self._update_log(f'Loop frequency updated to {value:.2f}Hz.')
+        self.run_callbacks('LoopFreq-RB', value)
+        return True
+
+    def set_control_qs(self, value):
+        """Set whether to include QS or not in feedforward."""
+        if not 0 <= value < len(_ETypes.DSBL_ENBL):
+            return False
+        self._control_qs = value
+        act = ('En' if value else 'Dis')
+        self._update_log(f'{act}abled QS control.')
+        self.run_callbacks('ControlQS-Sts', value)
+        return True
+
+    def set_config_name(self, value):
+        """Set configuration name."""
+        if self._loop_state == self._const.LoopState.Closed:
+            self._update_log('ERR:Open loop before changing configuration.')
+            return False
+
+        if not self._load_config(value):
+            return False
+
+        self._config_name = value
+        self.run_callbacks('ConfigName-RB', value)
+        return True
+
+    def _load_config(self, config_name):
+        try:
+            self._idff.load_config(config_name)
+            self._update_log(f'Updated configuration: {config_name}.')
+        except ValueError as err:
+            self._update_log('ERR:'+str(err))
+            return False
+        return True
+
+    #  ----- log auxiliary methods -----
+
+    def _update_log(self, msg):
+        if 'ERR' in msg:
+            _log.error(msg[4:])
+        elif 'FATAL' in msg:
+            _log.error(msg[6:])
+        elif 'WARN' in msg:
+            _log.warning(msg[5:])
+        else:
+            _log.info(msg)
+        self.run_callbacks('Log-Mon', msg)
+
+    # ----- auto save methods -----
+
+    def read_autosave_file(self):
```

Comment:
I never liked the idea of autosave implemented in local filesystem as the initial state of the IOC depends on where it runs (production in container/storage vs local test instantiations). I always thought configdb should we used for this (as this service should always be available!).

But if we will keep with local filesystem read/write, should not these functions be genralized and moved to a general section in siriuspy, so that all IOCs use the same read/save parameterized methods?

Reply:
Just like other (not pcaspy) IOCs, autosave is a feature that is usuful for a set of devices, so the user do not need to resend all configurations when IOC is restarted. 

> should not these functions be genralized and moved to a general section in siriuspy, so that all IOCs use the same read/save parameterized methods?

maybe, I see we have been using autosave for some IOCs, but I don't think this is the scope of this PR. Maybe we could unify the autosave feature in a future PR.

Reply:
yes, I know it is a feature. but in my view it is a design flaw. there should be an epics service for this. in our case we have configdb